build/.vsts-PRInternalChecks-azureBuild-pipeline.yml

```diff
@@ -0,0 +1,150 @@
+name: pr$(System.PullRequest.PullRequestNumber)-$(Date:yyyyMMdd)$(Rev:-r)
+parameters:
+- name: BuildConfiguration
+  default: release
+- name: NugetSecurityAnalysisWarningLevel
+  default: warn   
+variables:
+- name: vmImage
+  value: windows-2019
+resources:
+  repositories:
+  - repository: self
+    type: git
+    ref: main
+jobs:
+
+- job: Phase_1
+  displayName: PR Internal Checks for fhir-server - build and restore phase
+  cancelTimeoutInMinutes: 1
+  pool:
+    vmImage: $(vmImage)
+
+  steps:
+  - checkout: self
+    clean: true
+  - task: NuGetToolInstaller@0
+    displayName: Use NuGet 4.3.0
+    enabled: False
+    inputs:
+      versionSpec: 4.3.0
+
+  - task: NuGetCommand@2
+    displayName: NuGet restore
+    enabled: False
+    inputs:
+      selectOrConfig: config
+      nugetConfigPath: nuget.config
+
+  - task: UseDotNet@2
+    displayName: Use .NET Core sdk
+    inputs:
+      inputs:
+      version: 5.0.x
+      useGlobalJson: true
+      installationPath: $(Agent.ToolsDirectory)/dotnet
```

Comment:
I think the only input needed here is the useGlobalJson: true. At least the second input: is not needed.